core/lib/dal/src/eth_sender_dal.rs

```diff
@@ -86,6 +86,25 @@ impl EthSenderDal<'_, '_> {
         Ok(count.try_into().unwrap())
     }
 
+    pub async fn get_inflight_txs_count(&mut self) -> DalResult<usize> {
+        let count = sqlx::query!(
+            r#"
+            SELECT
+                COUNT(*)
+            FROM
+                eth_txs
+            WHERE
+                confirmed_eth_tx_history_id IS NULL
+            "#
+        )
```

Comment:
This query can be confusing as we use "inflight" for txs that are not yet confirmed and have at least one entry in eth_txs_history. Maybe let's use `get_unconfirmed_txs_count`?